src/components/ImageUpload.tsx

```diff
@@ -1,65 +1,98 @@
+import { StyledFilledButton, StyledPaperInput } from "./StyledComponents";
 
-interface ImageUploadProps {
-  setFieldValue: (field: string, value: any, shouldValidate?: boolean | undefined) => void;
-  errors: FormikErrors<{
-    description: string;
-    genre: string;
-    imageFile: {
-      name: string;
-      size: string;
-      type: string;
-    };
-  }>;
-  touched: FormikTouched<{
-    description: string;
-    genre: string;
-    imageFile: {
-      name: string;
-      size: null;
-      type: string;
-    };
-    releaseDate: string;
-    title: string;
-    yourRole: string;
-  }>;
-}
+// interface ImageUploadProps {
+//   setFieldValue: (field: string, value: any, shouldValidate?: boolean | undefined) => void;
+//   errors: FormikErrors<{
+//     description: string;
+//     genre: string;
+//     imageFile: {
+//       name: string;
+//       size: string;
+//       type: string;
+//     };
+//   }>;
+//   touched: FormikTouched<{
+//     description: string;
+//     genre: string;
+//     imageFile: {
+//       name: string;
+//       size: null;
+//       type: string;
+//     };
+//     releaseDate: string;
+//     title: string;
+//     yourRole: string;
+//   }>;
+// }
+
+export const ImageUpload = () => {
+  const PUBLIC_CLOUDINARY_CLOUD_NAME = "projectnewm",
+    PUBLIC_CLOUDINARY_UPLOAD_PRESET = "rvktckuk";
+
+  const [uploadedFilePublicId, setUploadedFilePublicId] = useState<string>();
+  const onDrop = useCallback(acceptedFiles => {
+    const url = `https://api.cloudinary.com/v1_1/${PUBLIC_CLOUDINARY_CLOUD_NAME}/upload`;
+
+    acceptedFiles.forEach(async (acceptedFile: File) => {
+      const formData = new FormData();
+      formData.append("file", acceptedFile);
+      formData.append("upload_preset", PUBLIC_CLOUDINARY_UPLOAD_PRESET);
+
+      const response = await fetch(url, {
+        body: formData,
+        method: "post",
+      });
+
+      const data = await response.json();
+      setUploadedFilePublicId(data.public_id);
+    });
+  }, []);
 
-export const ImageUpload = (props: ImageUploadProps) => {
-  const { setFieldValue } = props;
   const { getRootProps, getInputProps, isDragActive } = useDropzone({
     accept: "image/*",
-    onDrop: acceptedFiles => {
-      setFieldValue("files", acceptedFiles);
-    },
+    multiple: false,
+    onDrop,
   });
   return (
     <Box>
-      { }
       <div { ...getRootProps({ className: "dropzone" }) }>
-        <input { ...getInputProps() } />
-        <StyledPaperInput
-          sx={ {
-            width: "326px",
-            height: "38px",
-          } }
-        >
-          { " " }
-          Drag & Drop{ " " }
-        </StyledPaperInput>
-
-        <Typography variant="body1" align="center" color="primary">
-          OR
-        </Typography>
-        { isDragActive ? <p>Drop the files here ...</p> : <p>Drag n drop some files here, or click to select files</p> }
+        { !uploadedFilePublicId ? (
```

Comment:
nit: ternary is a bit hard to read. I think a simple `if` with early return might make it more clear

Reply:
Agreed. I will change.